src/main/java/org/mineskin/MineskinClient.java

```diff
@@ -185,6 +185,34 @@ private Connection getRequest(String endpoint) {
         }, requestExecutor);
     }
 
+    /**
+     * Uploads and generates skin data from a local file (with default options)
+     */
+    public CompletableFuture<Skin> generateUpload(File file) throws FileNotFoundException {
+        return generateUpload(file, SkinOptions.none());
```

Comment:
should keep the previous behaviour of it using the file's name by default (L179)